les/lespay/payment/lotterypmt/lottery_payment.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package lotterypmt
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/contracts/lotterybook"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/les/lespay/payment"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Identity is the unique string identity of lottery payment.
+const Identity = "Lottery"
+
+// RevealPeriod is the full life cycle length of lottery. The
+// number is quite arbitrary here, it's around 6.4 hours. We
+// can set a more reasonable number later.
+const RevealPeriod = 5760
+
+var errInvalidOpt = errors.New("invalid operation")
+
+// Role is the role of user in payment route.
+type Role int
```

Comment:
Is there a reason to use the same `Manager` for different roles? I don't see a lot of shared code, even the shared functions behave differently. `Config` also only has `Role` in it which does not seem to make a lot of sense. If you need a config in the future that would probably also be different for senders and receivers.

Reply:
Done.